Renderer.cs

```diff
@@ -65,61 +161,84 @@ public static void Initialize(GraphicsDevice gd, ContentManager content)
-            fonts = new List<SpriteFont>();
+            fonts = new List<IFont>();
         else
             fonts.Clear();
 
+        fontSystem = new FontSystem();
         string originalContentRoot = contentManager.RootDirectory;
 
-#if XNA
-        var contentManagerType = contentManager.GetType();
-        var rootDirectoryField = contentManagerType.GetField("rootDirectory", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.SetField | BindingFlags.GetField);
-        var fullRootDirectoryField = contentManager.GetType().GetField("fullRootDirectory", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.SetField | BindingFlags.GetField);
-#endif
-
         foreach (string searchPath in AssetLoader.AssetSearchPaths)
         {
-            string newRootDirectory = SafePath.GetDirectory(searchPath).FullName;
+            string baseDir = SafePath.GetDirectory(searchPath).FullName;
+            string iniPath = Path.Combine(baseDir, "Fonts.ini");
 
-            while (true)
+            if (File.Exists(iniPath))
             {
-                string sfName = string.Format(CultureInfo.InvariantCulture, "SpriteFont{0}", fonts.Count);
-
-                if (!SafePath.GetFile(searchPath, FormattableString.Invariant($"{sfName}.xnb")).Exists)
-                    break;
-
-#if !XNA
-                contentManager.RootDirectory = newRootDirectory;
-#else
-                // XNA does not allow changing the value of RootDirectory after the
-                // content manager has been used. However, it has some internal fields
-                // we can modify through reflection to achieve the same.
+                var iniFile = new IniFile(iniPath);
+                int fontCount = iniFile.GetIntValue("Fonts", "Count", 0);
+
+                for (int i = 0; i < fontCount; i++)
+                {
+                    string section = $"Font{i}";
+                    string fontPath = iniFile.GetStringValue(section, "Path", "");
+                    int size = iniFile.GetIntValue(section, "Size", 16);
+                    string fontTypeStr = iniFile.GetStringValue(section, "Type", nameof(FontType.SpriteFont));
+
+                    if (!Enum.TryParse<FontType>(fontTypeStr, true, out var fontType))
+                        fontType = FontType.SpriteFont;
+
+                    switch (fontType)
+                    {
+                        case FontType.TrueType:
+                            string fullFontPath = SafePath.GetFile(searchPath, fontPath).FullName;
+                            if (File.Exists(fullFontPath))
+                            {
+                                fontSystem.AddFont(File.ReadAllBytes(fullFontPath));
+                                fonts.Add(new TTFFontWrapper(fontSystem.GetFont(size)));
+                            }
+                            else
+                                Logger.Log($"TTF font file not found: {fullFontPath}");
+                            break;
```

Comment:
Don't mix braced and braceless blocks in the same if statement.

```suggestion
                            else
                            {
                                Logger.Log($"TTF font file not found: {fullFontPath}");
                            }
                            break;
```

Reply:
Fixed.